Documentation/contributors/design-proposals/webhooks.md

```diff
@@ -0,0 +1,125 @@
+    spec:
+      clusterPermissions:
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - "*"
+            ...
+      permissions:
+        - serviceAccountName: my-operator
+          rules:
+          - apiGroups:
+            - ""
+            ...
+      deployments:
+        - name: webhook-deployment
+          spec:
+            replicas: 1
+            selector:
+              matchLabels:
+                name: mywebhook
+            template:
+              metadata:
+                labels:
+                  name: mywebhook
+              spec:
+                serviceAccountName: my-operator
+                containers:
+                - name: webhook
+                  image: operator-framework/fake-webhook-image:latest
+                  imagePullPolicy: IfNotPresent
+                  ports:
+                  - containerPort: 443
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                - name: operator
+                  image: operator-framework/fake-operator:latest
+                  imagePullPolicy: IfNotPresent
+                  env:
+                  - name: NAMESPACE
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.annotations['olm.targetNamespaces']
+                  - name: MY_POD_NAME
+                    valueFrom:
+                      fieldRef:
+                        fieldPath: metadata.name
+      webhooks:
+      - name: olm-validation
+        type: ValidatingAdmissionWebhook
+        spec:
+          rules:
+          - apiGroups:
+            - "operators.olm.com"
+            apiVersions:
+            - v1alpha1
+            operations:
+            - create
+            resources:
+            - clusterserviceversions
+          selector:
+            name: mywebhook
+          webhookPath: "/admit"
+      - name: olm-mutation
+        type: MutatingAdmissionWebhook
+        spec:
+        ...
+
+Notable omissions in the above example include:
+- volume and volumeMounts in the deployment for the certificates
+- services
+- clientConfig in the webhooks section (path was retained via webhookPath)
+
+### Securing webhook via certificates
+
+It is planned for OLM to handle certificate creation for securing webhook connection via CSRs (`CertificateSigningRequest`). When a CSR is issued, the API server uses its CA certificate to sign a certificate for consumption and in this case will be used for the webhook. By default the `ClusterSigningDuration` is set to be [one year](https://github.com/kubernetes/kubernetes/blob/c48ed1c5aca6e06b6a3d70d5f189216df3150379/pkg/controller/certificates/signer/config/v1alpha1/defaults.go#L44) and is set via a flag to the kube-controller-manager called "--experimental-cluster-signing-duration duration".
```

Comment:
What happens to the cert after the duration? Does olm need to create another CSR, or is there a mechanism to auto-request a new one?

Reply:
(This is why this document is still marked as WIP.) OLM will need to create another CSR and there is no auto-renew mechanism like there is for kubelet certificates. In addition, we'll have to delete the old CSR and approve the new CSR as well.

Reply:
I've removed this section after changing directions on CSR usage.